src/fork_gc.c

```diff
@@ -328,8 +363,10 @@ static void FGC_childCollectNumeric(ForkGC *gc, RedisSearchCtx *sctx) {
         array_free(valuesDeleted);
       }
 
-      // we are done with the current field
-      FGC_sendPtrAddr(gc, 0);
+      if (header.sentFieldName) {
+        // If we've repaired at least one entry, send the terminator
```

Comment:
I think it might be a good idea to explain here why we send zero and not terminator (also to explain that its intentional.)